rest/route/task.go

```diff
@@ -203,6 +215,20 @@ func (h *updateArtifactURLHandler) Parse(ctx context.Context, r *http.Request) e
 	if err := evergreenutil.CheckURL(h.body.NewURL); err != nil {
 		return gimlet.ErrorResponse{StatusCode: http.StatusBadRequest, Message: fmt.Sprintf("new_url invalid: %s", err.Error())}
 	}
+
+	h.isSignedURL = strings.Contains(h.body.CurrentURL, "Token=")
+	if h.isSignedURL {
+		currentBucket, currentFileKey := parseS3URL(h.body.CurrentURL)
+		newBucket, newFileKey := parseS3URL(h.body.NewURL)
+
+		if currentBucket == "" || newBucket == "" || currentBucket != newBucket {
```

Comment:
Should we do a different error message for currentBucket != newBucket to avoid confusion?